### PR TITLE
[MM-56320] Create test cases for user limit warnings

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Dev PR: https://github.com/mattermost/mattermost/pull/25797

https://mattermost.atlassian.net/browse/MM-56320

### CODE REVIEW

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+Prerequisites:
+
+1. Have 1 Free instance running Enterprise edition with 9001 users running as an airgapped instance
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:40]: 10K is the limit https://docs.google.com/document/d/1RYL-6WrZdMjZYL84PvS_PsRdcnU3gXVCAOSIz5zl8Do/edit?usp=sharing

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Steps:
+
+**Scenario**: When user is running enterprise edition
+
+1. Login into Free instance running Enterprise edition with 9001 users.
+2. Verify a red global banner should be displayed with a message saying `Your user count is close to the maximum user limit. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Contact sales` button.
+3. Click on `Contact sales` button.
+4. Verify a popup with heading `Looks like you do not have access to the internet` is displayed.
+5. Add Professional/Enterprise license to the instance.
+6. Verify the global banner is not showing up.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:52]: this should be another scenario it cannot happen in a single test case

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+**Step 1**
+
+Prerequisites:
+
+1. Have 1 Free instance running Enterprise edition with 9001 users running as an airgapped instance
+2. Have 1 admin account.
+
+Steps:
+
+**Scenario**: When user is running enterprise edition
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:45]: this is for admins not users

Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:45]:
```suggestion
**Scenario**: When admin is running enterprise edition without license
```

File: data/test-cases/self-hosted/user-limits/Airgapped.md (hunk 2 above)
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:52]: another case " when admin is running enterprise edition with license"
- should not show the banner when limit is breached

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Please recheck the tests because we are not implementing the bands limit anymore check the spec https://docs.google.com/document/d/1RYL-6WrZdMjZYL84PvS_PsRdcnU3gXVCAOSIz5zl8Do/edit?usp=sharing

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+Scenario: When user is running enterprise edition
+
+1. Login into Free instance running Enterprise edition.
+2. Add a new user to the instance to make total active user to 10000.
+3. Verify a Red global banner should be displayed with a message saying `Your user count exceeds the maximum users allowed. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Contact sales` button.
+4. Navigate to System Console > Edition and License page.
+5. Verify a section with header `Upgrade to Mattermost Professional or Mattermost Enterprise` is shown in red background.
+6. Add Professional/Enterprise license to the instance.
+7. Verify the global banner is not showing up.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:55]: lets add refresh after the license added as well

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Scenario: When user is running team edition
+
+1. Login into Free instance running Team edition.
+2. Add a new user to the instance to make total active user to 10000.
+3. Verify a Red global banner should be displayed with a message saying `Your user count exceeds the maximum users allowed. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Contact sales` button.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:61]: before this refresh

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+4. Navigate to System Console > Edition and License page.
+5. Verify a section with header `Upgrade to Mattermost Professional or Mattermost Enterprise` is shown in red background.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:63]: not in the spec

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+Steps:
+
+**Scenario**: When instance is running on an Enterprise edition without license
+
+1. Login into Free instance running Enterprise edition with 10,000 users.
+2. Verify a red global banner should be displayed with a message saying `Your user count is close to the maximum user limit. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Contact sales` button.
+3. Click on `Contact sales` button.
+4. Verify a popup with heading `Looks like you do not have access to the internet` is displayed.
+5. Add Professional/Enterprise license to the instance.
+6. Verify the global banner is not showing up.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:52]: refresh the app

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+**Scenario**: When instance is running on an Enterprise edition with license
+
+1. Login into Free instance running Enterprise edition with 10,000 users.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:56]: not free instance

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md (hunk 4 above)
Comment (yasserfaraazkhan) [reply data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:55]: currently, after license the banners automatically disappear after adding license eg: the trial banner

File: data/test-cases/self-hosted/user-limits/Airgapped.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+2. Verify a red global banner should be displayed with a message saying `Your user count is close to the maximum user limit. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Contact sales` button.
+3. Click on `Contact sales` button.
+4. Verify a popup with heading `Looks like you do not have access to the internet` is displayed.
+
+**Scenario**: When instance is running on an Enterprise edition with license
+
+1. Login into instance running Enterprise edition with 10,000 users.
+2. Refresh the app and verify the banner is not displayed when user crosses maximum user limit.
```
Comment (M-ZubairAhmed) [inline data/test-cases/self-hosted/user-limits/Airgapped.md:55]: add scenario for air gapped team edition as well,

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
Thanks @yasserfaraazkhan! LGTM

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
+---
+# (Required) Ensure all values are filled up
+name: "Mattermost Enterprise and Team Edition should show a non dismissable global banner when instance has active users more than 10,000"
```
Comment (saturninoabril) [inline data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:3]: I suggest to omit "Mattermost" in the title for it's understandable already. In addition, I believe it should be "non-dismissible" although I can see we're not consistent as well.

```suggestion
name: "Enterprise and Team Edition should show a non-dismissible global banner when instance has active users more than 10,000"
```

File: data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+4. Have 1 Free instance running Enterprise edition with 10k users.
+5. Have 1 Free instance running Team edition with 10k users.
+
+Steps:
+
+**Scenario**: When user is running Enterprise edition.
+
+1. Login into Free instance running Enterprise edition.
+2. Add a new user to the instance to make total active user to 10000.
+3. Refresh the app and verify a Red global banner should be displayed with a message saying `Your user count exceeds the maximum users allowed. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Learn more` button.
```
Comment (saturninoabril) [inline data/test-cases/self-hosted/user-limits/BannerMoreThan10k.md:53]: Minor correction, please do the same to other similar usage.

```suggestion
3. Refresh the app and verify a red global banner is displayed with a message saying `Your user count exceeds the maximum users allowed. Upgrade to Mattermost Professional or Mattermost Enterprise to continue using Mattermost.`. There should also be a `Learn more` button.
```

**Review by saturninoabril — APPROVED** [review]
Thanks Yasser! Left minor comments, once resolve all good to me. Approving now.

**Pull request merged**